client/my-sites/email/form/mailboxes/components/new-mailbox-list/index.tsx

```diff
@@ -0,0 +1,185 @@
+
+	// Set visibility for desired hidden fields
+	mailboxes.forEach( ( mailbox ) => {
+		hiddenFieldNames.forEach( ( fieldName ) => {
+			mailbox.setFieldIsVisible( fieldName, false );
+			mailbox.setFieldIsRequired( fieldName, false );
+		} );
+	} );
+
+	const addMailbox = () => {
+		const newMailboxes = [ ...mailboxes, createNewMailbox() ];
+		setMailboxes( newMailboxes );
+		const eventName = isTitan
+			? 'calypso_email_titan_add_mailboxes_add_another_mailbox_button_click'
+			: 'calypso_email_google_workspace_add_mailboxes_add_another_mailbox_button_click';
+		recordTracksEvent( eventName, { mailbox_count: newMailboxes.length } );
+	};
+
+	const removeMailbox = useCallback(
+		( uuid: string ) => () => {
+			const newMailboxes = mailboxes.filter(
+				( mailbox ) => mailbox.formFields.uuid.value !== uuid
+			);
+			setMailboxes( newMailboxes );
+			const eventName = isTitan
+				? 'calypso_email_titan_add_mailboxes_remove_mailbox_button_click'
+				: 'calypso_email_google_workspace_add_mailboxes_remove_mailbox_button_click';
+			recordTracksEvent( eventName, { mailbox_count: newMailboxes.length } );
+		},
+		[ isTitan, mailboxes ]
+	);
+
+	const handleCancel = () => onCancel();
+
+	const persistMailboxesToState = () => {
+		setMailboxes( [ ...mailboxes ] );
+	};
+
+	const handleSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		mailboxes.forEach( ( mailbox ) => mailbox.validate( true ) );
+		persistMailboxesToState();
+
+		onSubmit( new MailboxOperations( mailboxes, persistMailboxesToState ) );
+	};
+
+	return (
+		<form onSubmit={ handleSubmit }>
+			<fieldset className="new-mailbox-list__main" disabled={ areButtonsBusy }>
+				{ mailboxes.map( ( mailbox, index ) => {
+					const uuid = mailbox.formFields.uuid.value;
+
+					const onFieldValueChanged = ( field: MailboxFormFieldBase< string > ) => {
+						if ( ! [ FIELD_FIRSTNAME, FIELD_NAME ].includes( field.fieldName ) ) {
+							return;
+						}
+						if ( mailbox.getIsFieldTouched( FIELD_MAILBOX ) ) {
+							return;
+						}
+
+						mailbox.setFieldValue( FIELD_MAILBOX, sanitizeMailboxValue( field.value ) );
+						mailbox.validateField( FIELD_MAILBOX );
+
+						mailbox.formFields.mailbox.dispatchState();
+					};
+
+					return (
+						<Fragment key={ 'form-' + uuid }>
+							{ index > 0 && (
+								<CardHeading
+									className="new-mailbox-list__numbered-heading"
+									tagName="h3"
+									size={ 20 }
+								>
+									{ translate( 'Mailbox %(position)s', {
+										args: { position: index + 1 },
+										comment:
+											'%(position)s is the position of the mailbox in a list, e.g. Mailbox 1, Mailbox 2, etc',
+									} ) }
+								</CardHeading>
+							) }
+							<MailboxFormWrapper mailbox={ mailbox } onFieldValueChanged={ onFieldValueChanged }>
+								<>
+									<div className="new-mailbox-list__actions">
+										{ index > 0 && (
+											<Button onClick={ removeMailbox( uuid ) } busy={ areButtonsBusy }>
```

Comment:
Minor UX thing, but maybe we should just disable this instead of passing `busy`. It looks a little bit strange when this button indicates that it's working after I've submit the form.

Reply:
I don't know that it's strange, as it's consistent with most UIs in the section.

Also IMO, isn't the `disabled` attribute mostly used to signify features that aren't available?

Reply:
The behavior of the current form implementation is that only the "Continue" button gets the busy animation (see screencast for quick reference). I would argue that this makes sense from a UX perspective, as that animation signals to me that "this particular operation is running". 

As for the `disabled` attribute – I agree, but that's also the point as I see it 🙂 While the form is submitting, none of the inputs or buttons should be available to the user. 

https://user-images.githubusercontent.com/1101677/173052024-a17bb122-f41b-49d5-9ca9-e5e50122e403.mov

Reply:
I think it's very important to signal to the user that there is an ongoing operation, and we should do that with the active submission button, which is `Continue` in this case.

For the other action buttons, which aren't active at the time - I agree with you on that front.

Reply:
Fully agree that the Continue button should have the busy animation!